ironfish-rust/src/serializing.rs

```diff
@@ -90,40 +88,25 @@ pub(crate) mod aead {
     ///
     /// This is just a facade around the ChaCha20Poly1305 struct. It ignores
     /// nonce and aad and automatically stores the mac tag.
-    pub(crate) fn encrypt(key: &[u8], plaintext: &[u8], encrypted_output: &mut [u8]) {
-        assert_eq!(encrypted_output.len(), plaintext.len() + MAC_SIZE);
-        let mut encryptor = ChaCha20Poly1305::new(key, &[0; 8], &[0; 8]);
-        let mut tag = [0; MAC_SIZE];
-        encryptor.encrypt(
-            plaintext,
-            &mut encrypted_output[..plaintext.len()],
-            &mut tag,
-        );
-        encrypted_output[plaintext.len()..].clone_from_slice(&tag);
+    /// This is just a facade around the ChaCha20Poly1305 struct. The nonce and
+    /// associated data are zeroed.
+    pub(crate) fn encrypt(key: &[u8; 32], plaintext: &[u8]) -> Result<Vec<u8>, IronfishError> {
+        let encryptor = ChaCha20Poly1305::new(Key::from_slice(key));
+        let result = encryptor
+            .encrypt(&Nonce::default(), plaintext)
+            .map_err(|_| IronfishError::InvalidSigningKey)?;
+        Ok(result)
```

Comment:
We could probably avoid one layer of unwraps if we don't return `Vec<u8>` here. I think we can use [`encrypt/decrypt_in_place_detached`](https://docs.rs/chacha20poly1305/latest/chacha20poly1305/trait.AeadInPlace.html#tymethod.encrypt_in_place_detached) to work with arrays directly. This removes the need to call `try_into()` as a follow-up to these functions

```rust

    pub(crate) fn encrypt<const SIZE: usize>(
        key: &[u8; 32],
        plaintext: &[u8],
    ) -> Result<[u8; SIZE], IronfishError> {
        let mut encrypted_output = [0u8; SIZE];
        encrypted_output[..plaintext.len()].copy_from_slice(plaintext);

        let encryptor = ChaCha20Poly1305::new(Key::from_slice(key));

        let tag = encryptor
            .encrypt_in_place_detached(
                &Nonce::default(),
                &[],
                &mut encrypted_output[..plaintext.len()],
            )
            .map_err(|_| IronfishError::InvalidSigningKey)?;

        encrypted_output[plaintext.len()..].copy_from_slice(&tag);

        Ok(encrypted_output)
    }

    pub(crate) fn decrypt<const SIZE: usize>(
        key: &[u8; 32],
        ciphertext: &[u8],
    ) -> Result<[u8; SIZE], IronfishError> {
        let decryptor = ChaCha20Poly1305::new(Key::from_slice(key));

        let mut plaintext = [0u8; SIZE];
        plaintext.copy_from_slice(&ciphertext[..SIZE]);

        decryptor
            .decrypt_in_place_detached(
                &Nonce::default(),
                &[],
                &mut plaintext,
                ciphertext[SIZE..].into(),
            )
            .map_err(|_| IronfishError::InvalidDecryptionKey)?;

        Ok(plaintext)
    }
```